Safety/AttitudeManager/Src/attitudeStateClasses.cpp

```diff
@@ -101,10 +115,24 @@ attitudeState& sensorFusionMode::getInstance()
 
 void PIDloopMode::execute(attitudeManager* attitudeMgr)
 {
-
-    CommandsForAM *PMInstructions = fetchInstructionsMode::GetPMInstructions();
+    CommandsForAM *PMInstructions = nullptr;
     SFOutput_t *SFOutput = sensorFusionMode::GetSFOutput();
 
+    PID_Output_t *pidOut = nullptr;
+    if(fetchInstructionsMode::isAutonomous())
+    {
+        PMInstructions = fetchInstructionsMode::GetPMInstructions();
+        //TODO: Run controls module with PMInstructions
+    }
+    else
+    {
+        PPM_Instructions_t *teleopInstructions = fetchInstructionsMode::GetTeleopInstructions();
+        _PidOutput = getPIDFromControls(teleopInstructions, SFOutput);
+    }
+
+    #ifdef FIXED_WING
+    PMInstructions = fetchInstructionsMode::GetPMInstructions();
+    
     //executes PID's to acheive desired roll, pitch angle
     //if manual control is needed, use loaded in percents instead!
     if(PMInstructions->passbyData.pitchPassby)
```

Comment:
this stuff uses the old PID loop correct?

Reply:
If so, it'll have to change to accomodate Anni's control module for testing since the old PID.hpp technically exists on Autopilot only iirc. @antholuo would you know how to use your controls module instead?

Reply:
Yeah so all we need to do would be changing the `_PidOutput.throttlePercent` to match the current PidOutput type which is defined in AM....

From controls we have:
`    if (PID_method == 0) {
        // use the base pid written without extensive scaling.

        // currents are all 0 because we reset the frame with reference to the quadcopter.

        pid_x = x_pid.execute(translate_lon, 0);
        pid_y = - y_pid.execute(translate_lat, 0); // global and airframe positive left/right are flipped
        pid_a = a_pid.execute(dist_alt, 0);
        pid_h = h_pid.execute(angl_hdn, 0);
    } else if (PID_method == 1) {
        // use a simple p-loop integrator
    } else {
        // not sure yet
    }

    PID_out.frontLeftMotorPercent = - pid_x - pid_y + pid_a + pid_h; // not sure about pid_h;
    PID_out.frontRightMotorPercent = - pid_x + pid_y + pid_a - pid_h; 
    PID_out.backLeftMotorPercent = pid_x - pid_y + pid_a - pid_h;
    PID_out.backRightMotorPercent = pid_x + pid_y + pid_a + pid_h;

    // to return PID (double check the return type?)
    return &PID_out;`

which is already returning motor Percents,  so we should just need to match them with the proper names?